### PR TITLE
Update gengo vendored dir

### PR DESCRIPTION
**Pull request by nikhita**

### CODE REVIEW

**Comment by nikhita** [comment]
```
➜  kubernetes git:(defaulter-gen-vendor) hack/verify-godeps.sh                                                                                                                                            0s 
Checking for 'Godeps/' changes against 'upstream/master'
Starting to download all kubernetes godeps. This takes a while
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/equality): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/errors): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/meta): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/resource): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/testing): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/validation): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/api/validation/path): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apimachinery): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apimachinery/announced): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apimachinery/registered): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apis/meta/internalversion): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apis/meta/v1): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apis/meta/v1/unstructured): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apis/meta/v1/validation): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/apis/meta/v1alpha1): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/conversion): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/conversion/queryparams): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/conversion/unstructured): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/fields): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/labels): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/openapi): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/schema): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer/json): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer/protobuf): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer/recognizer): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer/streaming): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer/versioning): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/runtime/serializer/yaml): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/selection): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/types): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/cache): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/clock): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/diff): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/errors): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/framer): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/httpstream): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/httpstream/spdy): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/intstr): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/json): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/jsonmergepatch): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/mergepatch): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/net): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/rand): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/remotecommand): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/runtime): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/sets): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/strategicpatch): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/uuid): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/validation): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/validation/field): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/wait): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/util/yaml): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/version): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/pkg/watch): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/third_party/forked/golang/json): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/third_party/forked/golang/netutil): unable to determine git status exit status 128
# cd /tmp/gopath.cThmKq/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery; git status
fatal: Not a git repository (or any of the parent directories): .git
godep: error downloading dep (k8s.io/apimachinery/third_party/forked/golang/reflect): unable to determine git status exit status 128
godep: Error downloading some deps. Aborting restore and check.
!!! Error in hack/verify-godeps.sh:93
  Error in hack/verify-godeps.sh:93. 'godep restore' exited with status 1
Call stack:
  1: hack/verify-godeps.sh:93 main(...)
Exiting with status 1
Removing /tmp/gopath.cThmKq
```

**Comment by nikhita** [comment]
Basically this ends up destroying the symlink at `vendor/k8s.io/apimachinery` and generates all the code (not respecting the symlink).

This is especially weird to me because I didn't mention `k8s.io/apimachinery` anywhere while using godep. :thinking: 

Moreover, `hack/godep-save.sh` makes sure symlinks are proper: https://github.com/kubernetes/kubernetes/blob/master/hack/godep-save.sh#L50. So this should not happen...

**Comment by nikhita** [comment]
What I did (mirrors the instructions from this doc - https://github.com/kubernetes/community/blob/master/contributors/devel/godep.md):

1. Devoting a separate directory to this endeavor:

```bash
export KPATH=$HOME/code/kubernetes
mkdir -p $KPATH/src/k8s.io
cd $KPATH/src/k8s.io
git clone https://github.com/nikhita/kubernetes.git   # this is at HEAD
```

2. Set up GOPATH.

```bash
# This will *not* let local builds see packages that exist elsewhere on my system.
export GOPATH=$KPATH
```

3. Populate new GOPATH.

```bash
cd $KPATH/src/k8s.io/kubernetes
./hack/godep-restore.sh
```

4. Update k8s.io/gengo using:

```bash
cd $KPATH/src/k8s.io/kubernetes
DEP=k8s.io/gengo
rm -rf $KPATH/src/$DEP
godep get $DEP/...
rm -rf Godeps
rm -rf vendor
./hack/godep-save.sh
```

This creates the extra directory `vendor/github.com/docker/docker/project/` which I removed manually.

```bash
rm -rf vendor/github.com/docker/docker/project/
./hack/update-bazel.sh
./hack/update-godep-licenses.sh
```

Then commit the changes + update the staging area:

```bash
./hack/update-staging-client-go.sh
./hack/verify-godeps.sh  # generates an error
```

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -2906,44 +2913,280 @@
 			"Rev": "53feefa2559fb8dfa8d81baad31be332c97d6c77"
 		},
 		{
+			"ImportPath": "k8s.io/apimachinery/pkg/api/equality",
```
Comment (sttts) [inline Godeps/Godeps.json]: these shouldn't be here, strange

Comment (nikhita) [reply Godeps/Godeps.json]: Yes...

**Pull request closed without merging**